### ISSUE TITLE
TravisCI making it look like it works

### ISSUE THREAD

**Issue by isaac-weisberg** [khanlou/Promise#36]
TravisCI is as commonly known doesn't work with Swift because of some issues with multi targeted Objective-C builds in Xcode or whatever...
And also it has come to my attention that in 7b9764c397f8ba992cb4103d9e900d3d885c5e90 the Travis badge has been removed altogether, which makes me think that its absence is commonly tolerated and accepted :)

Suggestion: @khanlou should log in at public https://travis-ci.org and disable automatic builds for this repository altogether, so it doesn't show that build failed for literally every single commit and pull-request.

**Comment by isaac-weisberg** [comment]
So I've looked at them logs
```
	“Use Legacy Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.

```
So did anyone try doing that?

**Comment by isaac-weisberg** [comment]
So I have been able to configure .travis.yml to support generic language processing and it really is working in #40, both Linux and iOS. Didn't bother configuring for macOS since it's the same as on iOS. Shoutout to @colemancda for helping me out with sample config :+1: 

Now, the trick with the whole thing is that it works only with fully configured SPM tests on Linux. Now, we have indeed configured them tests in #38 and there is no doubt that the configuration works with it since #40 is already built on top of that PR, with tests configured. So essentially, until #38 is merged, the solution will still be not functional.

**Comment by colemancda** [comment]
Cool! Glad to be of help. Just out of curiosity which commit / project exactly did you see that helped you (I have been using the same Travis CI config in a couple projects).

**Comment by isaac-weisberg** [comment]
@colemancda one of your latest jams https://github.com/PureSwift/Codable !

**Comment by khanlou** [comment]
@isaac-weisberg I'm not really able to follow this, because I don't use this on Linux with SPM. And you are right, Travis went from flaky to unreliable so I dropped the badge for it. If you can solve both of these issues and recommend changes to me I would be open to merging that. Let me know what the status of it is!

**Comment by isaac-weisberg** [comment]
@khanlou  tests are solved in #38 , Travis is solved in #41 . #41 will not really work without #38 since I've wrote the config based on contents of #38 and Travis will expect SPM tests from #38 too, so it's preferred to merge #38 first.
